web/src/lib/utils/file-uploader.ts

```diff
@@ -209,3 +209,23 @@ function handleUploadError(asset: File, respBody?: any) {
 		timeout: 5000
 	});
 }
+
+async function mergeMap<T>(source: (() => Promise<T>)[], concurrent = 2) {
+	const results: T[] = [];
+
+	while (source.length) {
+		concurrent = concurrent > source.length ? source.length : concurrent;
+
+		const selectedFns = source.splice(0, concurrent);
+		const selectedResults = await Promise.all(selectedFns.map((fn) => fn()));
+
+		results.push(...selectedResults);
+	}
+
+	return results;
+}
+
+function getFileExt(asset: File) {
```

Comment:
Why don't we use this method to get extension name? https://nodejs.org/api/path.html#pathextnamepath

Reply:
NodeJS function on the web app? the svelte support polyfills for this? The important thing is that someone has already coded it, I just copy and create the function to share.

Reply:
Ah my bad